File: Projet_final/angularBT/src/app/model/stagiaire.ts
import { Cursus } from './cursus';
import { Ordinateur } from './ordinateur';
import { User } from './user';
export class Stagiaire extends User{
  
   constructor(private _ordinateur:Ordinateur=null,
  private _cursus:Cursus=null) {
     super();

  }
  
  public get ordinateur(): Ordinateur {
    return this._ordinateur;
  }
  
  public set ordinateur(_ordinateur: Ordinateur) {
    this._ordinateur =_ordinateur;
  }
  
  public get cursus(): Cursus {
    return this._cursus;
  }
  
  public set cursus(_cursus: Cursus) {
    this._cursus =_cursus;
  }
  
}

File: Projet_final/projetFinalSpringBoot/src/main/java/com/exemple/model/Stagiaire.java
package com.exemple.model;

import javax.persistence.DiscriminatorValue;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.OneToOne;

import com.fasterxml.jackson.annotation.JsonView;

@Entity
@DiscriminatorValue("Stagiaire")
public class Stagiaire extends User {
	@OneToOne
	@JsonView(JsonViews.UserWithOrdi.class)
	private Ordinateur ordinateur;
	@ManyToOne(fetch = FetchType.LAZY)
	@JoinColumn(name = "Stagiaire_cursus")
	@JsonView(JsonViews.UserWithCursus.class)
	private Cursus cursus;

	public Ordinateur getOrdinateur() {
		return ordinateur;
	}

	public void setOrdinateur(Ordinateur ordinateur) {
		this.ordinateur = ordinateur;
	}

	public Cursus getCursus() {
		return cursus;
	}

	public void setCursus(Cursus cursus) {
		this.cursus = cursus;
	}
}
